### PR TITLE
refactor: contributor workflow to reuse common parts

### PR DESCRIPTION
**Pull request by OleksandrNechai**
### Description

Extract all the logic into composite GitHub action.

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

### CODE REVIEW

**Review by denieler — APPROVED** [review]
👍

**Comment by denieler** [comment]
please do the same for Davinci 🙇  😄

**Review by TomasSlama — APPROVED** [review]
This looks pretty cool! Reusable github action! LGTM 👍

File: .github/workflows/handle-contribution.yml [hunk 1]
```diff
@@ -1,40 +1,28 @@
...
-              :pray: please, help us improve, [rate your contributing experience](https://forms.gle/unzJBEJAHRz7LEtX8) after merge
-              `
-            })
+          repository: toptal/davinci-github-actions
+          token: ${{ secrets.TOPTAL_DEVBOT_TOKEN }}
+          ref: 'gha/notify-jira'
+          path: ./.github/actions/
+
+      - uses: ./.github/actions/notify-jira-about-contribution
+        if: github.event.pull_request.draft == false && !(github.actor == 'dependabot[bot]')
```
Comment (LashaJini) [inline .github/workflows/handle-contribution.yml:22]: Ignore `draft` and `dependeabot` PRs.

File: .github/workflows/handle-contribution.yml [hunk 2]
```diff
@@ -1,40 +1,28 @@
 name: Handle contribution
 
 on:
-  workflow_dispatch:
```
Comment (LashaJini) [inline .github/workflows/handle-contribution.yml:4]: We don't need to manually dispatch this workflow. Can't think of edge cases when we should.

File: .github/workflows/handle-contribution.yml [hunk 3]
```diff
@@ -1,40 +1,28 @@
...
   pull_request:
-    types: [opened]
+    types: [opened, reopened, ready_for_review]
```
Comment (LashaJini) [inline .github/workflows/handle-contribution.yml:5]: PR should be ready for review when we switch from `draft` to ready for review, and when we `reopen` closed PR.

**Pull request closed without merging**